### ISSUE TITLE
Shrink gmtmex.c by growing gmtmex_parser.c to be included in GMT core

### ISSUE THREAD

**Issue by PaulWessel** [GenericMappingTools/gmtmex#12]
**Description of the desired feature**

The gmtmex._extension_ library with the mexFunction needs to be built using the user's MATLAB version.  Hence, we want to move as much of the code in gmtmex.c into the gmtmex_parser.c as possible to keep things simple. It seems to me that the ultimate mexFunction would just be something like this:

```
#include "gmtmex.h"
void mexFunction (int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[]) {
	gmt_mexfunction (nlhs, plus, nrhs, prhs);
	return;
}
```

Does gmtmex.c really need more in it, @joa-quim if we just place the rest in gmt_mexfunction and stick in the gmtmex_parser.c in libgmt?

**Comment by PaulWessel** [comment]
The benefit of the above is you can write a short paragraph to tell people just make this 5-line file and compile it with max or similar.

**Comment by joa-quim** [comment]
I don't think you can shrink it that much. If you move any ``mxArray`` (or family) to the GMT lib than the entire GMT would depend on Matlab.

**Comment by PaulWessel** [comment]
Hm, well gmtmex_parser.c calls mexErrMsgTxt, mxCreateNumericMatrix and lots of things, so it is all requiring a matlab include file, so perhaps there is nothing we can supply that does not require mex include/lib?

**Comment by joa-quim** [comment]
Yes, had forgotten about those. So, nope it seems there is nothing valuable that we can move.

**Comment by PaulWessel** [comment]
I will look but it reduces the scope.
